SeleniumAutomationGenerator/ComponentsFactory.cs

```diff
@@ -2,148 +2,65 @@
-
-            return GenerateFileCreatorOutputs(selector, keyWord, children, filteredChildren);
+            _creatorsSteps.Add(step);
+            return this;
         }
 
-        private IEnumerable<ComponentGeneratorOutput> GenerateFileCreatorOutputs(string selector, string keyWord, IEnumerable<AutoElementData> children, AutoElementData[] filteredChildren)
+        public IEnumerable<ComponentGeneratorOutput> CreateCsOutput(string body)
         {
-            return GenerateAppendersOutputs(keyWord, children)
-                            .Union(GenerateFileCreatorOutputs(selector, filteredChildren, keyWord), new ComponentOutputComparer());
+            IEnumerable<AutoElementData> children = AutoElementFinder.GetChildren(body);
+            return GenerateClassesForElements(children);
         }
 
-        private IEnumerable<ComponentGeneratorOutput> GenerateAppendersOutputs(string keyword, IEnumerable<AutoElementData> elements)
+        public IEnumerable<ComponentGeneratorOutput> InvokeStep(AutoElementData current,
+            IComponentFileCreator parentClassCreator, IComponentFactoryCreatorStep next = null)
         {
-            IComponentFileCreator parent = _fileCreatorContainer.GetFileCreator(keyword);
-            IEnumerable<AutoElementData> appendersElements = elements
-                            .Where(IsAppenderElement);
-            return GenerateClassesForElements(appendersElements, parent);
-        }
+            _appendersSteps
+                .Where(step => step.ShouldInvokeStep(current, parentClassCreator))
+                .ForEach(step => step.InvokeStep(current, parentClassCreator));
 
-        private void RunCustomAttributesAppends(AutoElementData current, IComponentFileCreator parentClassCreator)
-        {
-            current.AutoAttributes
-                .Where(_attributesContainer.ContainsCustomAttribute)
-                .Select(_attributesContainer.GetElementAttribute)
-                .ForEach(att => RunAppender(parentClassCreator, att, current));
+            var result = _creatorsSteps.FirstOrDefault(step => step.ShouldInvokeStep(current, parentClassCreator))
+                ?.InvokeStep(current, parentClassCreator, this);
+            return result ?? new List<ComponentGeneratorOutput>();
         }
 
-        private IEnumerable<ComponentGeneratorOutput> GenerateAppenderOutputs(AutoElementData current,
-            IComponentFileCreator parentClassCreator, string keyWord, IEnumerable<AutoElementData> filteredChildren)
+        public bool ShouldInvokeStep(AutoElementData rootElement, IComponentFileCreator parent)
         {
-            RunAppender(parentClassCreator, _classAppenderContainer.GetAppender(keyWord), current);
-            return GenerateClassesForElements(filteredChildren);
+            return true;
         }
 
         private IEnumerable<ComponentGeneratorOutput> GenerateClassesForElements(IEnumerable<AutoElementData> children, IComponentFileCreator parent = null)
```

Comment:
this method is only called once without the second parameter. It does not need the parameter `IComponentFileCreator parent = null`